pkg/resource/endpoint_config/testdata/test_suite.yaml

```diff
@@ -1,17 +1,119 @@
+                message: "The endpoint config name must not include a special character."
+        invoke: Create
+        expect:
+          latest_state: "v1alpha1/create/observed/invalid_create_attempted.yaml"
+          error: resource is in terminal condition
+      - name: "Create=Valid"
+        description: "Create a new endpoint config successfully (ARN in status)."
+        given:
+          desired_state: "v1alpha1/create/desired/success_before_create.yaml"
+          svc_api:
+            - operation: CreateEndpointConfigWithContext
+              output_fixture: "sdkapi/create/create_success.json"
+        invoke: Create
+        expect:
+          latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+  - name: "Endpoint config readOne tests"
+    description: "Testing the readOne operation"
+    scenarios:
+      - name: "ReadOne=MissingRequiredField"
+        description: "Testing readOne when required field is missing. No API call is made and returns error."
+        given: 
+          desired_state: "v1alpha1/readone/desired/missing_required_field.yaml"
+        invoke: ReadOne
+        expect:
+          error: "resource not found"
+      - name: "ReadOne=NotFound"
+        description: "Testing readOne when Describe fails to find the resource on SageMaker"
+        given: 
+          desired_state: "v1alpha1/readone/desired/right_after_create.yaml"
+          svc_api:
+            - operation: DescribeEndpointConfigWithContext
+              error:
+                code: ValidationException
+                message: "Could not find endpoint configuration single-variant-config"
+        invoke: ReadOne
+        expect:
+          error: "resource not found"
+      - name: "ReadOne=Fail"
+        description: "This test checks if the condition is updated if describe fails and readOne returns error"
+        given: 
+          desired_state: "v1alpha1/readone/desired/right_after_create.yaml"
+          svc_api:
+            - operation: DescribeEndpointConfigWithContext
+              error:
+                code: ServiceUnavailable
+                message: "Server is down"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/error_on_describe.yaml"
+          error: "ServiceUnavailable: Server is down\n\tstatus code: 0, request id: "
+      - name: "ReadOne=AfterCreate"
+        description: "Testing readOne right after create, there should be no delta."
+        given:
+          desired_state: "v1alpha1/readone/desired/right_after_create.yaml"
+          svc_api:
+            - operation: DescribeEndpointConfigWithContext
+              output_fixture: "sdkapi/describe/describe_success.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/desired/right_after_create.yaml"
+      - name: "ReadOne=SuccessClearsConditions"
+        description: "Testing successful api call after an error right clears the conditions."
```

Comment:
nit: Unclear description

Reply:
yes reworded, it tests that a successful reconciliation clears conditions if  terminal/recoverable condition is already set